assets/src/components/v2/departures/later_departures.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import _ from "lodash";
+import cx from "classnames";
+import weakKey from "weak-key";
+
+import { hasOverflowX } from "Util/util";
+
+import DepartureRow from "./departure_row";
+import { Row } from "./normal_section";
+import RoutePill from "./route_pill";
+
+const MAX_LATER_DEPARTURES = 5;
+
+const LaterDepatures = ({ rows }: { rows: Row[] }) => {
+  const [limit, setLimit] = useState(MAX_LATER_DEPARTURES);
+  const departures = useMemo(
+    () =>
+      _.take(
+        rows.filter(
+          (r): r is DepartureRow & { type: "departure_row" } =>
+            r.type === "departure_row",
+        ),
+        limit,
+      ),
+    [rows, limit],
+  );
+
+  const ref = useRef<HTMLDivElement>(null);
+  const [idx, setIdx] = useState(0);
+
+  useLayoutEffect(() => {
+    if (hasOverflowX(ref)) {
+      setLimit(Math.max(0, limit - 1));
+    }
+  }, [limit]);
+
+  useEffect(() => {
+    if (departures.length < 2) return;
+    const refreshMs = Math.min(3750, 15000 / departures.length);
+
+    const interval = setInterval(() => {
+      setIdx((i) => (i + 1) % departures.length);
+    }, refreshMs);
+
+    return () => clearInterval(interval);
+  }, [departures.length]);
+
+  const departure = departures[idx % departures.length];
+
+  return (
+    <div className="later-departures">
+      <div className="later-departures__header" ref={ref}>
+        <h3>Later Departures</h3>
+        <ol>
+          {departures.map((d) => (
+            <li
+              key={weakKey(d)}
+              className={cx({
+                "later-departures--departure--selected": d === departure,
+              })}
+            >
+              <RoutePill {...d.route} outline={d !== departure} />
+            </li>
+          ))}
+        </ol>
+      </div>
+      <div className="later-departures__departures">
```

Comment:
Maybe we should name this `__departure` or `__current-departure`, since it only displays one? (this also appears un-referenced in the CSS, not sure if intentional)